### PR TITLE
Target Sum Triplets in Golang

### PR DESCRIPTION
**Pull request by GajjarMiten**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes, I have read the Contributing Guidelines on Pull Requests.

### Description

- Added implementation of the program for Target Sum Triplets in Golang 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #585  for Go language

### CODE REVIEW

**Comment by GajjarMiten** [comment]
Hey @iamrajiv  @Kajol-Kumari @SKAUL05 
@siddharth25pandey @plazzy99  @atarax665 
could you please review this pull request?

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@GajjarMiten add time and space complexity. Also, this program belongs to `cp` section not in `ds` section.

**Comment by GajjarMiten** [comment]
> @GajjarMiten add time and space complexity. Also, this program belongs to `cp` section not in `ds` section.

@iamrajiv 
I have made changes. Could you please review it?

**Review by atarax665 — CHANGES_REQUESTED** [review]
Looks good to me, just remove unnecessary line breaks.

File: Go/cp/target_sum_triplets.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+input:
+	9
+	5 7 9 1 2 4 6 8 3
+	10
+	output:
+	1, 2 and 7
+	1, 3 and 6
+	1, 4 and 5
+	2, 3 and 5
+
```
Comment (iamrajiv) [inline Go/cp/target_sum_triplets.go]: @GajjarMiten remove this line break

File: Go/cp/target_sum_triplets.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+					// if the sum is less than targetSum we should
+					// increase the left index for the next element
+					left++
+				} else {
+					// if the sum is greater than targetSum we should
+					// decrease the right index for the next element
+					right--
+				}
+			}
+
```
Comment (iamrajiv) [inline Go/cp/target_sum_triplets.go]: remove this line break

File: Go/cp/target_sum_triplets.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		}
+	}
+	// if there is no triplet exist
+	if count == 0 {
+		fmt.Println("Sorry couldn't found any triplets with target sum")
+	}
+
```
Comment (iamrajiv) [inline Go/cp/target_sum_triplets.go]: remove this line break

File: Go/cp/target_sum_triplets.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	sumTriplets(array, targetSum)
+
+}
+
+/*
+
+	Time Complexity : O(n^2) because of two nested loops
+	Space Complexity : O(1)
+
```
Comment (iamrajiv) [inline Go/cp/target_sum_triplets.go:101]: remove this line break

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: Go/cp/target_sum_triplets.go (hunk 4 above)
Comment (GajjarMiten) [reply Go/cp/target_sum_triplets.go:101]: > remove this line break

@iamrajiv 
I have removed unnecessary linebreaks
kindly check it out.

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :)

**Review by iamrajiv — APPROVED** [review]
@GajjarMiten LGTM!

**Comment by GajjarMiten** [comment]
> Looks good to me, just remove unnecessary line breaks.

Hey @atarax665 
I have made the changes.
could you please review this :)

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Pull request merged**